chainerui/models/result.py

```diff
@@ -43,17 +44,60 @@ def __init__(self, path_name=None, name=None, project_id=None,
     def __repr__(self):
         return '<Result id: %r, path_name: %r />' % (self.id, self.path_name)
 
-    @property
-    def serialize(self):
-        """serialize."""
+    @classmethod
+    def create(cls, path_name=None, name=None, project_id=None,
+               log_modified_at=None):
+        """Initialize an instance and save it to db."""
+        result = cls(path_name, name, project_id, log_modified_at)
+
+        DB_SESSION.add(result)
+        DB_SESSION.commit()
+
+        from chainerui.tasks import crawl_result
+        crawl_result(result.id, True)
+
+        return result
+
+    def sampled_logs(self, logs_limit=-1):
+        """Return up to `logs_limit` logs.
+
+        If `logs_limit` is -1, this function will return all logs that belong
+        to the result.
+        """
+        logs_count = len(self.logs)
+        if logs_limit == -1 or logs_count <= logs_limit:
+            return self.logs
+        elif logs_limit == 0:
+            return []
+        elif logs_limit == 1:
+            return [self.logs[logs_count - 1]]
+        else:
+            def get_sampled_log(idx):
+                # always include the first and last element of `self.logs`
+                return self.logs[idx * (logs_count - 1) // (logs_limit - 1)]
+            return [get_sampled_log(i) for i in range(logs_limit)]
```

Comment:
`test_api.py` checks this sampling logic as count of log on GET, but does not check that the content of log array is correct or not. For example,  `len(logs) is 4` and `logs_limig is 3`, `test_api.py` passes `[logs[0], logs[1], logs[2]]`,  `[logs[0], logs[1], logs[3]]` is correct, however, followed to the comment, I think.

I think those tests prefer to be added on `models_test/test_result.py` which does not exist yet, not on `test_api.py`

Reply:
I created `models_test/test_result.py` and added some test cases to it.